subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestOptionsIntegrationSpec.groovy

```diff
@@ -0,0 +1,360 @@
+        |   options {
+        |       includeTags 'fast'
+        |   }
+        |}
+        |
+        |testing {
+        |   suites {
+        |       test {
+        |           useJUnit() // NOT ALLOWED, task is already configured with a framework, should fail-fast here
+        |       }
+        |   }
+        |}""".stripMargin()
+
+        when:
+        fails ":test"
+
+        then:
+        result.assertHasErrorOutput("You cannot set the test framework on suite: test after accessing test options on an associated Test task: test.")
+    }
+
+    def "can set non-options test property for test task in #suiteName prior to setting framework within suite"() {
+        given:
+        resources.maybeCopy("TestOptionsIntegrationSpec")
+        buildFile << """
+        |testing {
+        |   suites {
+        |       $suiteDeclaration {
+        |           targets {
+        |               all {
+        |                   testTask.configure {
+        |                       minHeapSize = "128m"
+        |                   }
+        |               }
+        |           }
+        |           useJUnitJupiter()
+        |           targets {
+        |               all {
+        |                   testTask.configure {
+        |                       options {
+        |                           ${type} 'fast'
+        |                       }
+        |                   }
+        |               }
+        |           }
+        |       }
+        |   }
+        |}""".stripMargin()
+
+        expect:
+        succeeds ":$task"
+
+        where:
+        suiteName   | suiteDeclaration              | task        | type
+        'test'      | 'test'                        | 'test'      | 'includeTags'
+        'test'      | 'test'                        | 'test'      | 'excludeTags'
+        'integTest' | 'integTest(JvmTestSuite)'     | 'integTest' | 'includeTags'
+        'integTest' | 'integTest(JvmTestSuite)'     | 'integTest' | 'excludeTags'
+    }
+
+    // See JUnitCategoriesIntegrationSpec for the inspiration for this test
+    def "re-executes test when #type is changed in #suiteName"() {
+        given:
+        resources.maybeCopy("TestOptionsIntegrationSpec")
+        buildFile << """
+        |testing {
+        |   suites {
+        |       $suiteDeclaration {
+        |           useJUnitJupiter()
+        |           targets {
+        |               all {
+        |                   testTask.configure {
+        |                       options {
+        |                           ${type} 'fast'
+        |                       }
+        |                   }
+        |               }
+        |           }
+        |       }
+        |   }
+        |}""".stripMargin()
+
+        when:
+        succeeds ":$task"
+
+        then:
+        executedAndNotSkipped ":$task"
+
+        when:
+        resources.maybeCopy("TestOptionsIntegrationSpec")
+        buildFile << """
+        |testing {
```

Comment:
Not sure how I feel about the `|` in the blocks here. I can imagine this may make searching the codebase more difficult. I can think of a few times I've used regex to search around (including text with newlines) and I can imagine this getting in the way. 

Is there an advantage I'm missing here?

Reply:
I usually don't add them and fully left-align just to keep the tests visually cleaner.  The test in `JUnitCategoriesIntegrationSpec` that led me down this rabbit hole used them, so I did too.  I don't have strong feelings either way.  I'll remove them.